### PR TITLE
allow release and next versions to be specified on the command line

### PR DESCRIPTION
**Pull request by blake-boesinger**
This pull request would resolve this issue: https://github.com/sbt/sbt-release/issues/96

I found the parsing library to be a bit strange, resulting in args being a Seq[Serializable], the elements are  either ((String, List[String]),String) or String.

In 'readVersion', I have defaulted to using the command line version, if absent, then use the default version, if absent, then prompt for versions.

### CODE REVIEW

**Comment by jroper** [comment]
Could you add a scripted test to test this?

File: src/main/scala/ReleasePlugin.scala [hunk 1]
```diff
@@ -103,17 +108,35 @@ object ReleasePlugin extends AutoPlugin {
...
+      private val releaseVersionParser = Space ~> releaseVersionStringParser ~ Space ~ StringBasic
+      private val nextVersionParser = Space ~> nextVersionStringParser ~ Space ~ StringBasic
+      private val releaseParser = (releaseVersionParser | nextVersionParser | Space ~> WithDefaults | Space ~> SkipTests | Space ~> CrossBuild).*
+
+      val releaseCommand: Command = Command(releaseCommandKey)(_ => releaseParser) { (st, args: Seq[Serializable]) =>
         val extracted = Project.extract(st)
         val releaseParts = extracted.get(releaseProcess)
         val crossEnabled = extracted.get(releaseCrossBuild) || args.contains(CrossBuild)
+
+        //args is a Seq[Serializable], the elements are  either ((String, List[String]),String) or String
```
Comment (xuwei-k) [inline src/main/scala/ReleasePlugin.scala]: `Seq[Serializable]` is terrible

https://github.com/xuwei-k/sbt-release/commit/160f856b26e643ee57

**Comment by blake-boesinger** [comment]
Tests added. Apologies, I wasn't aware there was a testing framework for sbt or what scripted test was :)

**Comment by mlenner** [comment]
+1

**Comment by svinci** [comment]
Will it be merged sometime in the future?

**Comment by damdev** [comment]
+1

**Comment by dyoodespegar** [comment]
+1

**Comment by fggarcia** [comment]
+1

**Comment by fgiannetti** [comment]
+1

**Comment by cristianspinetta** [comment]
+1

**Comment by juliccr** [comment]
+1 @svinci

**Comment by mreparaz** [comment]
+1 @svinci

**Comment by crypticmind** [comment]
+1

**Comment by smagnacco** [comment]
+1

**Comment by guscampo** [comment]
+1 @svinci

**Comment by avinocur** [comment]
+1

**Comment by mjuanes** [comment]
+1 I ❤ @svinci

**Comment by victorbetran** [comment]
+1

**Comment by Rick0** [comment]
+1

**Comment by jroper** [comment]
This appears as one email thread in my gmail inbox, so only one person needs to ask a question for me to notice - getting everyone in your company to +1 it makes no difference to whether I notice it or not.  However, it does add a lot of noise to the pull request - the purpose of pull requests is to review the code, but with all this noise, how am I and the submitter of this PR supposed to follow the flow of review comments?  I want to make a comment to the submitter, but I have to scroll up to find out who they are, it's just annoying.  So, you're essentially just frustrating the process of getting this PR merged, and annoying everyone - doing that often causes maintainers to completely ignore your requests.  Fortunately for all of you, I'm not in a bad mood today, so I'm going to ignore all the +1's.

@blake-boesinger It looks like when you updated the PR, you brought in some rebased commits that are already on master - they don't change anything since they're already on master, but they will appear twice in the history which is not good.  The easiest way for you to fix this is to squash all the commits into one.

**Pull request merged**